src/org/traccar/protocol/Tk103ProtocolDecoder.java

```diff
@@ -190,11 +320,73 @@ private Position decodeNetwork(Channel channel, SocketAddress remoteAddress, Str
 
         getLastLocation(position, null);
 
+        getLastBatteryPower(position);
+
         position.setNetwork(new Network(CellTower.from(
                 parser.nextInt(0), parser.nextInt(0), parser.nextHexInt(0), parser.nextHexInt(0))));
 
         return position;
     }
+
+    private Position decodeLbsWifi(Channel channel, SocketAddress remoteAddress, String sentence) {
+        Parser parser = new Parser(PATTERN_LBSWIFI, sentence);
+        if (!parser.matches()) {
+            return null;
+        }
+
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
+        if (deviceSession == null) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        decodeType(position, parser.next(), "0");
+
+        getLastLocation(position, null);
+
+        getLastBatteryPower(position);
+
+        Network network = new Network();
+
+        // Parse LBS
+        network.addCellTower(CellTower.from(
+                parser.nextInt(), parser.nextInt(), parser.nextInt(), parser.nextInt()));
+
+        // Parse WiFi macs number and mac addresses itself.
+        int wifiCount = parser.nextInt();
+        if (parser.hasNext()) {
+            String[] wifimacs = parser.next().split(",");
+            if (wifimacs.length == wifiCount) {
+                for (int i = 0; i < wifiCount; i++) {
+                    // Sample wifi string: “00:80:E1:7F:86:97*-55*6” (mac*rssi*channel_number)
+                    String[] wifiinfo = wifimacs[i].split("\\*");
+                    network.addWifiAccessPoint(WifiAccessPoint.from(
+                            wifiinfo[0], Integer.parseInt(wifiinfo[1]), Integer.parseInt(wifiinfo[2])));
+                }
+            }
+        }
+
+        if (network.getCellTowers() != null || network.getWifiAccessPoints() != null) {
+            position.setNetwork(network);
+        }
+
+        // Parse date-time
+        DateBuilder dateBuilder = new DateBuilder()
```

Comment:
Why are you not using `parser.nextDateTime`?

Reply:
done